pkgs/tools/networking/openssh/ssh-keysign-8.5.patch

```diff
@@ -0,0 +1,24 @@
+diff --git a/pathnames.h b/pathnames.h
+index cb44caa4..354fdf05 100644
+--- a/pathnames.h
++++ b/pathnames.h
+@@ -124,7 +124,7 @@
+ 
+ /* Location of ssh-keysign for hostbased authentication */
+ #ifndef _PATH_SSH_KEY_SIGN
+-#define _PATH_SSH_KEY_SIGN		"/usr/libexec/ssh-keysign"
```

Comment:
Shouldn't we put `$out/bin/ssh-keysign` here eventually?

Reply:
I don't see any reason why we shouldn't, however I initially wanted to keep this PR as small as possible (just updating the package) without introducing any more changes.
This was not possible however because of the gssapi and hpn variants (and the review I got) so that's why the PR is so big. Would you be okay with keeping it as-is (it worked for years now) and revisiting that in a later commit? Not changing that would also make it a lot easier to backport this security-critical change because it mostly just updates the package and doesn't change any other behaviour.

Reply:
I checked the [man page](https://linux.die.net/man/8/ssh-keysign) and we can not do that:

> Since they are readable only by root, ssh-keysign must be set-uid root if host-based authentication is used.

So on NixOS we have to go through the wrappes and not through `$out` directly